### PR TITLE
Mark send_notifications as an ignored column

### PR DESCRIPTION
**Pull request by steventux**
## Context

Strong migrations ensures the column is ignored before it's physically removed.
This column will be removed in a subsequent migration.

## Changes proposed in this pull request

- Ignore `ProviderUser#send_notifications` column.
- Stop backfilling `ProviderUserNotificationPreferences` from `ProviderUser#send_notifications` value, just create the record when a user is created.
- Updates specs to setup with notification preferences instead of `send_notifications`

## Guidance to review

There's an implementation change here in that we were previously creating `ProviderUserNotificationPreferences` based on the value of `ProviderUser#send_notifications` but as this field will be removed we now just create a `ProviderUserNotificationPreferences` record associated with the provider user. See https://github.com/DFE-Digital/apply-for-teacher-training/pull/4879/commits/f122910ffdac6ce55151ebbc699a643f0d0d907e for this change.

## Link to Trello card

https://trello.com/c/04BJzJlM/3757-remove-old-notifications-from-database

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/factories/provider_user.rb [hunk 1]
```diff
@@ -65,5 +64,11 @@
         user.provider_permissions.update_all(view_diversity_information: true)
       end
     end
+
+    trait :with_notification_preferences_enabled do
```
Comment (benswannack) [inline spec/factories/provider_user.rb:68]: Minor, but maybe just `with_notifications_enabled`?

**Review by benswannack — APPROVED** [review]
This looks good to me, aside from a couple of minor things :)

File: spec/factories/provider_user.rb (hunk 1 above)
Comment (steventux) [reply spec/factories/provider_user.rb]: :+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4879/commits/29cd7473608fe89e70e99f994918a088711d227b

File: app/services/save_provider_user_notification_preferences.rb [hunk 2]
```diff
@@ -5,45 +5,18 @@ def initialize(provider_user:)
...
-    end
-  end
-
   def update_all_notification_preferences!(notification_preferences_params: {})
     return false if notification_preferences_params.empty?
 
     ActiveRecord::Base.transaction do
-      update_provider_user!(send_notifications_from_notificaion_prefernces(notification_preferences_params))
       provider_user_notification_preferences.update!(notification_preferences_params)
     end
```
[the remark this answers is not in the document]
Comment (steventux) [reply app/services/save_provider_user_notification_preferences.rb]: Removed / amended here https://github.com/DFE-Digital/apply-for-teacher-training/pull/4879/commits/b3ec38a4a47c9abe72b13ba7450f396f27b703f4

**Review by cwrw — COMMENTED** [review]
can we also remove https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/controllers/support_interface/provider_user_notification_preferences_controller.rb#L4 ?

**Review by cwrw — APPROVED** [review]
Looks good, just removing the last instance of backfill

**Comment by steventux** [comment]
@cwrw last mention of `backfill...` removed here https://github.com/DFE-Digital/apply-for-teacher-training/pull/4879/commits/a92013af82142e2cb57df7893ff23a703a2756fc

**Review by cwrw — APPROVED** [review]
thanks @steventux

**Pull request merged**